tests/phpunit/tests/test-switch-blog-query.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+if ( is_multisite() ) :
+
+	class Switch_Blog_Urls_Test extends PLL_Multisites_TestCase {
+		/**
+		 * @testWith ["http://example.org", "en"]
+		 *           ["http://example.org", "fr"]
+		 *
+		 * @param string $url  URL to test.
+		 * @param string $lang Current language slug.
+		 */
+		public function test_queries_blog_pll_dir( string $url, string $lang ) {
+			$this->clean_up_filters();
+
+			switch_to_blog( (int) self::$blog_with_pll_directory->blog_id );
+
+			$pll_frontend = $this->get_pll_frontend_env();
+			$pll_frontend->curlang = $pll_frontend->model->get_language( $lang ); // Force current language.
+
+			$url .= 'en' === $lang ? '' : "/$lang";
+
+			$this->go_to( $url );
+
+			$this->assertQueryTrue( 'is_home', 'is_front_page' );
+		}
+
+		/**
+		 * @testWith ["http://polylang-domains.en", "en"]
+		 *           ["http://polylang-domains.de", "de"]
+		 *
+		 * @param string $url URL to test.
+		 * @param string $lang Current language slug.
+		 */
+		public function test_queries_blog_pll_domains( string $url, string $lang ) {
+			$this->clean_up_filters();
+
+			switch_to_blog( (int) self::$blog_with_pll_domains->blog_id );
+
+			$pll_frontend = $this->get_pll_frontend_env();
+			$pll_frontend->curlang = $pll_frontend->model->get_language( $lang ); // Force current language.
+
+			$this->go_to( $url );
+
+			$this->assertQueryTrue( 'is_home', 'is_front_page' );
+		}
+
+		/**
+		 * @ticket #1855
+		 * @see https://github.com/polylang/polylang-pro/issues/1855.
+		 */
+		public function test_queries_blog_pll_dir_switched_twice() {
+			global $wp_rewrite;
+
+			$this->clean_up_filters();
+
+			switch_to_blog( (int) self::$blog_with_pll_directory->blog_id );
+
+			$pll_admin = $this->get_pll_admin_env();
+			do_action_ref_array( 'pll_init', array( &$pll_admin ) );
+
+			$post = $this->factory()->post->create();
+			$pll_admin->model->post->set_language( $post, 'fr' );
```

Comment:
What's the purpose? When I read the test title, we just need to switch the blog twice. No need to create a new post.

Reply:
I wondered the same thing. Removed in [7a6d02b](https://github.com/polylang/polylang/pull/1352/commits/7a6d02bfc731ee2a49db6313955b6ed855f031da).